src/service.ts

```diff
@@ -293,31 +287,26 @@ export class Service {
     return JSON.parse(await response.text());
   }
 
-  static saveJSON(json: object, uri: string): Promise<string> {
+  static async saveJSON(json: object, uri: string): Promise<any> {
     const update = !!uri;
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        if (update) {
-          resolve(uri);
-        } else {
-          let jsonURI = JSON.parse(this.response).uri;
-          jsonURI = jsonURI.substring(jsonURI.lastIndexOf("/") + 1);
-          resolve(jsonURI);
-        }
+    if (update) {
+      const url = "//api.myjson.com/bins/" + uri;
+      const response = await fetch(url, {
+        method: "put",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" })
       });
-      xhr.addEventListener("error", function() {
-        reject();
+      const json = JSON.parse(await response.text());
+      return Promise.resolve(uri);
+    } else {
+      const url = "//api.myjson.com/bins/";
+      const response = await fetch(url, {
+        method: "post",
+        headers: new Headers({ "Content-type": "application/json; charset=utf-8" }),
+        body: JSON.stringify(json)
       });
-      if (update) {
-        xhr.open("PUT", "//api.myjson.com/bins/" + uri, true);
-      } else {
-        xhr.open("POST", "//api.myjson.com/bins", true);
-      }
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
-    });
+      return JSON.parse(await response.text());
```

Comment:
return portion of `JSON.parse(await response.text()).uri;` as you see at removed code at line 304.